RequestService/RequestService.Repo/Repository.cs

```diff
@@ -190,44 +194,89 @@ public async Task<int> NewHelpRequestAsync(PostNewRequestForHelpRequest postNewR
 
                     foreach (HelpMyStreet.Utils.Models.Job job in postNewRequestForHelpRequest.NewJobsRequest.Jobs)
                     {
-
-                        EntityFramework.Entities.Job EFcoreJob = new EntityFramework.Entities.Job()
+                        if (job.SupportActivity.RequestType() == RequestType.Shift)
                         {
-                            NewRequest = newRequest,
-                            Details = job.Details,
-                            IsHealthCritical = job.HealthCritical,
-                            SupportActivityId = (byte)job.SupportActivity,
-                            DueDate = DateTime.Now.AddDays(job.DueDays),
-                            DueDateTypeId = (byte)job.DueDateType,
-                            JobStatusId = (byte)JobStatuses.New,
-                            Reference = job.Questions.Where(x => x.Id == (int)Questions.AgeUKReference).FirstOrDefault()?.Answer
-                        };
-                        _context.Job.Add(EFcoreJob);
-                        await _context.SaveChangesAsync();
-                        job.JobID = EFcoreJob.Id;
-
-                        foreach (var question in job.Questions)
+                            var locationQuestion = job.Questions.Where(x => x.Id == (int) Questions.Location).First();
+                            var numberOfSlotsQuestion = job.Questions.Where(x => x.Id == (int)Questions.NumberOfSlots).First();
+
+                            if(locationQuestion==null || numberOfSlotsQuestion == null )
+                            {
+                                throw new Exception("Missing location question or number of slots question");
+                            }
+
+                            int numberofSlots = Convert.ToInt32(numberOfSlotsQuestion.Answer);
+
+                            _context.Shift.Add(new EntityFramework.Entities.Shift()
+                            {
+                                Request = newRequest,
+                                StartDate = job.StartDate.Value,
+                                ShiftLength = (int) (job.EndDate.Value - job.StartDate.Value).TotalMinutes,
+                                LocationId = Convert.ToInt32(locationQuestion.Answer)
+                            });
+
+                            for (int i = 0; i < numberofSlots; i++)
+                            {
+                                EntityFramework.Entities.Job EFcoreJob = new EntityFramework.Entities.Job()
+                                {
+                                    NewRequest = newRequest,
+                                    IsHealthCritical = false,
+                                    SupportActivityId = (byte)job.SupportActivity,
+                                    DueDate = new DateTime(1900, 1, 1),
+                                    DueDateTypeId = (byte)DueDateType.SpecificStartAndEndTimes,
+                                    JobStatusId = (byte)JobStatuses.Open,
```

Comment:
For consistency, should we create these jobs as New and advance them to Open via an action?